src/aks-preview/azext_aks_preview/azurecontainerstorage/_helpers.py

```diff
@@ -0,0 +1,205 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.azclierror import UnknownError
+from azure.cli.command_modules.acs._roleassignments import (
+    add_role_assignment,
+    build_role_scope,
+    delete_role_assignments,
+)
+from azext_aks_preview._client_factory import get_providers_client_factory
+from azext_aks_preview.azurecontainerstorage._consts import (
+    CONST_ACSTOR_K8S_EXTENSION_NAME,
+    CONST_EXT_INSTALLATION_NAME,
+    CONST_K8S_EXTENSION_CLIENT_FACTORY_MOD_NAME,
+    CONST_K8S_EXTENSION_CUSTOM_MOD_NAME,
+    CONST_K8S_EXTENSION_NAME,
+    CONST_STORAGE_POOL_NAME_PREFIX,
+    CONST_STORAGE_POOL_OPTION_NVME,
+    CONST_STORAGE_POOL_RANDOM_LENGTH,
+    CONST_STORAGE_POOL_TYPE_ELASTIC_SAN,
+    CONST_STORAGE_POOL_TYPE_EPHEMERAL_DISK,
+    RP_REGISTRATION_POLLING_INTERVAL_IN_SEC,
+)
+
+from datetime import datetime
+from knack.log import get_logger
+import random
+import string
+import time
+
+logger = get_logger(__name__)
+
+
+def register_dependent_rps(cmd, subscription_id) -> bool:
+    required_rp = 'Microsoft.KubernetesConfiguration'
+    from azure.mgmt.resource.resources.models import ProviderRegistrationRequest, ProviderConsentDefinition
+
+    properties = ProviderRegistrationRequest(third_party_provider_consent=ProviderConsentDefinition(consent_to_authorization=False))
+    client = get_providers_client_factory(cmd.cli_ctx)
+    is_registered = False
+    try:
+        is_registered = _is_rp_registered(cmd, required_rp, subscription_id)
+        if is_registered:
+            return
+        client.register(required_rp, properties=properties)
+        # wait for registration to finish
+        timeout_secs = 120
+        start = datetime.utcnow()
+        is_registered = _is_rp_registered(cmd, required_rp, subscription_id)
+        while not is_registered:
+            is_registered = _is_rp_registered(cmd, required_rp, subscription_id)
+            time.sleep(RP_REGISTRATION_POLLING_INTERVAL_IN_SEC)
+            if (datetime.utcnow() - start).seconds >= timeout_secs:
+                logger.error("Timed out while waiting for the {0} resource provider to be registered.".format(required_rp))

```

Comment:
Does the while loop need a break statement?
```suggestion
        while not is_registered:
            is_registered = _is_rp_registered(cmd, required_rp, subscription_id)
            time.sleep(RP_REGISTRATION_POLLING_INTERVAL_IN_SEC)
            if (datetime.utcnow() - start).seconds >= timeout_secs:
                logger.error("Timed out while waiting for the {0} resource provider to be registered.".format(required_rp))
                break
```